File: message-bus/src/main/java/alma/icd/adapt/messagebus/configuration/RecipientGroupRepository.java
package alma.icd.adapt.messagebus.configuration;

import java.util.Optional;

import org.springframework.data.jdbc.repository.query.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Component;

import alma.icd.adapt.messagebus.rabbitmq.RecipientGroup;

@Component
public interface RecipientGroupRepository extends CrudRepository<RecipientGroup, String> {
	@Query( "select * " + 
			"from Recipient_Group " + 
			"where group_name = :groupName" )
	Optional<RecipientGroup> findByName( @Param("groupName") String name );
}

File: message-bus/src/main/python/adapt/messagebus/configuration/RecipientGroupRepository.py
class RecipientGroupRepository(object):
    #@Query( "select * " + 
    #        "from Recipient_Group " + 
    #        "where group_name = :groupName" )
    def findByName(self, name):
        return None
    def deleteAll(self):
        pass
        #raise NotImplementedError
    def findAll(self):
        return []
        #raise NotImplementedError
    def save(self, group):
        pass
